pkg/scheduler/predicates.go

```diff
@@ -255,6 +255,22 @@ func NewServiceAffinityPredicate(podLister PodLister, serviceLister ServiceListe
 	return affinity.CheckServiceAffinity
 }
 
+// TODO:@sdminonne: remove it once checked with @agupta
+func copyValueIfKeyEquals(key string, affinityLabels map[string]string, nodeSelector labels.Selector) bool {
```

Comment:
If you leave this, can you rework this into a "findValue(...)" method, that will make the code much more legible below:
```go
val, err := findValue(...)
if val != nil {
   affinityLabels[l] = val
}
...
```